server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -1,167 +1,117 @@
 package uk.gov.ons.addressIndex.server.modules
 
 import javax.inject.{Inject, Singleton}
-
 import uk.gov.ons.addressIndex.server.model.dao.ElasticClientProvider
 import com.google.inject.ImplementedBy
 import com.sksamuel.elastic4s.ElasticDsl._
 import com.sksamuel.elastic4s._
 import play.api.Logger
-import uk.gov.ons.addressIndex.crfscala.CrfScala.{CrfTokenResult, Input}
-import uk.gov.ons.addressIndex.model.db.index.{NationalAddressGazetteerAddress, NationalAddressGazetteerAddresses, PostcodeAddressFileAddress, PostcodeAddressFileAddresses}
-import uk.gov.ons.addressIndex.parsers.Tokens
-import uk.gov.ons.addressIndex.parsers.Tokens.Token
-
+import uk.gov.ons.addressIndex.crfscala.CrfScala.CrfTokenResult
+import uk.gov.ons.addressIndex.model.{AddressScheme, BritishStandard7666, PostcodeAddressFile}
+import uk.gov.ons.addressIndex.model.db.index.HybridIndex
+import uk.gov.ons.addressIndex.server.modules.Model._
 import scala.concurrent.{ExecutionContext, Future}
 
+object Model {
+  case class Pagination(offset: Int, limit: Int)
+  implicit class AutoPaginate(searchDefinition: SearchDefinition) {
+    def paginate(implicit p: Pagination): SearchDefinition = {
+      searchDefinition start p.offset limit p.limit
+    }
+  }
+
+  implicit class AutoSource(searchDefinition: SearchDefinition) {
+    def format(implicit optFmt: Option[AddressScheme]) = {
```

Comment:
It is implicit so the useage fits with the ElasticDSL.
(you dont have to add params in yourself, just apply the format which you have)
Checkout the impl lines:

`search.in(index).format`

keeps things clean